modules/lib/lib-portal/src/main/resources/lib/xp/portal.ts

```diff
@@ -600,3 +600,36 @@ export function imagePlaceholder(params: ImagePlaceholderParams): string {
     bean.setHeight(params?.height ?? 0);
     return bean.createImagePlaceholder();
 }
+
+export interface ApiUrlParams {
+    toolApplication?: string;
+    toolKey?: string;
```

Comment:
tool is implicit when requested from/admin 
In XP8 tools will unlikely be able to generate urls to each other as it is not secure. 
Exception is main tool (launcher) and wellcome - but they may have their own custom implementation